### PR TITLE
STIX spectrogram loader

### PR DESCRIPTION
**Pull request by natsat0919**
Code for STIX spectrogram data loading (addressing issue https://github.com/sunpy/sunkit-spex/issues/104 .). Currently this loader only supports processed STIX spectrogram files outputted by the `stx_convert_spectrogram` IDL function. A lot of the loader functions in StixLoader inherit from the old RhessiLoader class that was previously in the instruments.py file. I have added this loader to the legacy fitting folder instead of the extern folder as there will likely be changes to the instrument loaders in general. Once we decide on the format of the loader, we can move it to extern. 

Changes implemented:
Updated the STIX file loading functions
Added StixLoader (supports spectrogram data loading, choosing event and background times for spectral fitting, lightcurve and spectrogram plotting)

### CODE REVIEW

**Comment by KriSun95** [comment]
I think this looks good! Would it be worth while to move the code to the `extern` module, like where the RHESSI code is now?

**Comment by natsat0919** [comment]
> I think this looks good! Would it be worth while to move the code to the `extern` module, like where the RHESSI code is now?

Yeah sure, I will move it there. I just thought that if we plan on doing some refactoring, then it would be better to have it in the legacy folder.

**Comment by natsat0919** [comment]
@KriSun95 The STIX loader is now moved to extern!

File: changelog/160.feature.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Add STIX instrument loader
+
+- Add the STIX loader to :py:class:`sunkit_spex.extern.stix.StixLoader`
```
Comment (samaloney) [inline changelog/160.feature.rst:3]:
```suggestion
Add STIX instrument loader `~sunkit_spex.extern.stix.StixLoader`.
```

File: sunkit_spex/extern/stix.py [hunk 2]
```diff
@@ -0,0 +1,1393 @@
...
+import numpy as np
+
+from astropy.time import Time, TimeDelta
+from astropy import units as u
+from astropy.io import fits
+
+from sunkit_spex.legacy.fitting_legacy import instruments
+
+
+class StixLoader(instruments.InstrumentBlueprint):
```
Comment (samaloney) [inline sunkit_spex/extern/stix.py:14]: camel case for classes, snake case for functions, variable etc at 99% of the time
```suggestion
class STIXLoader(instruments.InstrumentBlueprint):
```

File: sunkit_spex/extern/stix.py [hunk 3]
```diff
@@ -0,0 +1,1393 @@
...
+    """
+    Loader specifically for processed STIX spectrogram data.
+
+    This loader has the same format as the STIX/RHESSI loader from the older version of sunkit-spex (sunxspex).
+    Once the format of the instrument loaders is decided, the loader should be moved to sunkit_spex.extern
```
Comment (samaloney) [inline sunkit_spex/extern/stix.py:19]: Update as in extern now?

File: sunkit_spex/extern/stix.py [hunk 4]
```diff
@@ -0,0 +1,1393 @@
...
+from astropy.time import Time, TimeDelta
+from astropy import units as u
+from astropy.io import fits
+
+from sunkit_spex.legacy.fitting_legacy import instruments
+
+
+class StixLoader(instruments.InstrumentBlueprint):
+    """
+    Loader specifically for processed STIX spectrogram data.
```
Comment (samaloney) [inline sunkit_spex/extern/stix.py:16]: Maybe say exported from the IDL software?

File: sunkit_spex/extern/stix.py [hunk 5]
```diff
@@ -0,0 +1,1393 @@
...
+    This loader has the same format as the STIX/RHESSI loader from the older version of sunkit-spex (sunxspex).
+    Once the format of the instrument loaders is decided, the loader should be moved to sunkit_spex.extern
+
+    StixLoader Specifics
+    ----------------------
+    Has methods to plot time series and perform time selection on the data. A background time can be added or removed and can fit
+    the event data with the model+background (recommended) or fit a model to data-background using the `data2data_minus_background`
+    setter with False or True, respectively.
+
+    We assume that the background (if calculated) and the event emission is calculated from the same sized area. If this is not
```
Comment (samaloney) [inline sunkit_spex/extern/stix.py:27]: More of question here is area here like a on disk area or a time interval?

File: sunkit_spex/extern/stix.py [hunk 6]
```diff
@@ -0,0 +1,1393 @@
...
+        -------
+        A 2d array of the counts [counts], count rates [counts/s], and the count and count
+        rate errors (counts, counts_err, cts_rates, cts_rate_err).
+        """
+        # The processed spectrogram files only come in counts/sec so need
+        if stix_dict["1"][0]["TTYPE1"] == "RATE":
+            cts_rates = stix_dict["1"][1]["RATE"]
+            cts_rate_err = stix_dict["1"][1]["STAT_ERR"]
+        else:
+            print("I don't know what units STIX has.")
```
Comment (samaloney) [inline sunkit_spex/extern/stix.py:291]: Prob better to raise and exception

**Review by samaloney — COMMENTED** [review]
Great stuff, few minor comments.

**Comment by natsat0919** [comment]
@samaloney thanks for the code review. I updated the code based on your comments. It should be ready to bed merged

**Comment by natsat0919** [comment]
I had to update the srm placeholder for the STIX loader, as custom placeholder names don't allow loading in of multiple files at the same time (since only pha_file, arf_file ... srm_file in the LoadSpec are actually allowed to be lists). Therefore, it was not possible to simultaneously load 2 STIX spectra at the same time for fitting

**Review by KriSun95 — APPROVED** [review]
Looks good to me. Thanks for moving stuff to `extern`!

**Review by samaloney — APPROVED** [review]

**Pull request merged**